### ISSUE TITLE
onSubmitted and onChanged appear to be broken

### ISSUE THREAD

**Issue by Sorunome** [AbdulRahmanAlHamali/flutter_typeahead#188]
In the text field configuration, onSubmitted and onChanged appear to be broken:

```
TextFieldConfiguration(
  onSubmitted: (String text) {
    // blank for now
  },
```
fails to compile with

```
Compiler message:
lib/components/input_bar.dart:35:22: Error: The argument type 'Null Function(String)' can't be assigned to
the parameter type 'void Function(dynamic)'.
        onSubmitted: (String text) {                                    
                     ^
```

If forcing with `as void Function(String)` at the end the same error appears as a runtime error.

When leaving out the `String` type definition things seem to work:
```
TextFieldConfiguration(
  onSubmitted: (text) {
    // use "text" as if it was a string
  },
```

Same applies for the onChanged property

**Comment by hillelcoren** [comment]
Looks like there's a merged PR for this: https://github.com/AbdulRahmanAlHamali/flutter_typeahead/pull/161

**Comment by Sorunome** [comment]
The issue still exists in 1.8.1, so with that PR, though

Am 15. Mai 2020 15:40:27 MESZ schrieb Hillel Coren <notifications@github.com>:

**Comment by hillelcoren** [comment]
I believe the PR was merged after 1.8.1, the changelog shows 2.0

https://github.com/AbdulRahmanAlHamali/flutter_typeahead/pull/161/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR1

**Comment by hillelcoren** [comment]
That said I haven't reviewed the code, too bad...
